packages/client/components/ActivityLibrary/CreateNewActivity/CreateNewActivity.tsx

```diff
@@ -0,0 +1,263 @@
+    const categoryId = params.categoryId
+    if (!categoryId) return defaultActivity
+
+    const selectedActivity = SUPPORTED_CUSTOM_ACTIVITIES.find((category) =>
+      category.includedCategories.includes(categoryId)
+    )
+    if (!selectedActivity) return defaultActivity
+    return selectedActivity
+  })
+  const [selectedTeam, setSelectedTeam] = useState(sortByTier(data.viewer.teams)[0]!)
+  const {submitting, error, submitMutation, onError, onCompleted} = useMutationProps()
+  const history = useHistory()
+
+  const handleCreateRetroTemplate = () => {
+    if (submitting) {
+      return
+    }
+
+    const teamTemplates = data.viewer.availableTemplates.edges.filter(
+      (template) => template.node.teamId === selectedTeam.id
+    )
+
+    if (teamTemplates.length >= Threshold.MAX_RETRO_TEAM_TEMPLATES) {
+      onError(new Error('You may only have 20 templates per team. Please remove one first.'))
+      return
+    }
+    if (teamTemplates.find((template) => template.node.name === '*New Template')) {
+      onError(new Error('You already have a new template. Try renaming that one first.'))
+      return
+    }
+
+    submitMutation()
+    AddReflectTemplateMutation(
+      atmosphere,
+      {teamId: selectedTeam.id},
+      {
+        onError,
+        onCompleted: (res: AddReflectTemplateMutation$data) => {
+          const templateId = res.addReflectTemplate?.reflectTemplate?.id
+          if (templateId) {
+            history.push(`/activity-library/details/${templateId}`, {
+              prevCategory: params.categoryId
+            })
+          }
+          onCompleted()
+        }
+      }
+    )
+  }
+
+  const handleCreatePokerTemplate = () => {
+    //TODO: implement me
+  }
+
+  const createCustomActivityLookup: Record<ActivityType, () => void> = {
+    retrospective: handleCreateRetroTemplate,
+    poker: handleCreatePokerTemplate
+  }
+
+  const handleActivitySelection = (activityType: ActivityType) => {
+    setSelectedActivity(
+      SUPPORTED_CUSTOM_ACTIVITIES.find((category) => category.type === activityType)!
+    )
+  }
+
+  return (
+    <div className='flex h-full w-full flex-col bg-white'>
+      <ActivityLibraryHeader
+        title='Create New Activity'
+        leftNavigation={
+          <Link className='p-4' to={`/activity-library/`} replace={true}>
+            <IconLabel icon={'arrow_back'} iconLarge />
+          </Link>
+        }
+      />
+      <div className='flex flex-1 flex-col items-center gap-y-8'>
+        <h1 className='text-lg font-normal'>
+          Choose an <span className='font-semibold'>Activity Format:</span>
+        </h1>
+        <RadioGroup.Root
+          className='mx-auto flex flex-col gap-8 sm:flex-row'
+          aria-label='Choose an Activity Format'
+          value={selectedActivity?.type}
+          onValueChange={handleActivitySelection}
+        >
+          {SUPPORTED_CUSTOM_ACTIVITIES.map((category) => {
+            return (
+              <RadioGroup.Item
```

Comment:
@enriquesanchez how should `focus` state look like for an item? Here's how it looks right now. It'll be used when someone navigates by keyboard
<img width="910" alt="Screenshot 2023-04-21 at 14 18 32" src="https://user-images.githubusercontent.com/1017620/233633567-1d8913d6-5377-4137-90ca-2825c628ed72.png">

Reply:
@BartoszJarocki Ideally the focus border should only appear on the thumbnail and not on the whole component. Would that be possible? Is the whole are clickable?

Reply:
> how it should look like when item is both selected & focused?

Great question! @ackernaut did you have anything in mind for these two separate states (selected and/or focused)? If not, I can spend time today looking into it so we don't delay Bartosz' work.

Reply:
I started exploring [some ideas over here](https://www.figma.com/file/PmE9A2ryODVzFPjZFJn7qi/Parabol-Activity-Library?node-id=1271-31494&t=6TNtzayvs4X78TKG-4). I'm looking at defining focus and hover states that can be consistent across all the activity library. 

@BartoszJarocki @ackernaut What do you think? I'm leaning towards option C

Reply:
C looks great!

Reply:
Gonna go and implement the following
<img width="1808" alt="CleanShot 2023-04-26 at 08 52 51@2x" src="https://user-images.githubusercontent.com/1017620/234493677-e53ccb41-930f-4a0e-85b1-c59c2fe5a700.png">

we can adjust later, for now it looks good imo!

Reply:
I took a look at Figma, and left comments there. Thanks for pushing forward on this @enriquesanchez !